bpf/bpf_sock.c

```diff
@@ -295,9 +295,11 @@ static __always_inline int __sock4_xlate_fwd(struct bpf_sock_addr *ctx,
 	struct lb4_service *svc;
 	__u16 dst_port = ctx_dst_port(ctx);
 	__u32 dst_ip = ctx->user_ip4;
+	__u32 proto = ctx->protocol;
 	struct lb4_key key = {
 		.address	= dst_ip,
 		.dport		= dst_port,
+		.proto		= (__u8)ctx->protocol,
```

Comment:
@borkmann is this correct? can I make this assumption to map the protocol from the context to an u8 directly?